openlibrary/catalog/marc/tests/test_get_subjects.py

```diff
@@ -114,10 +114,7 @@ def test_subjects_xml(self, item, expected):
     @pytest.mark.parametrize('item,expected', bin_samples)
     def test_subjects_bin(self, item, expected):
         filename = os.path.dirname(__file__) + '/test_data/bin_input/' + item
-
-        data = open(filename).read()
-        if len(data) != int(data[:5]):
-            data = data.decode('utf-8').encode('raw_unicode_escape')
+        data = open(filename, mode='rb').read()
```

Comment:
```suggestion
        with open(filename, mode='rb') as in_file
            data = in_file.read()
```
Close dangling file handles.